### DISCUSSION TITLE
Run phpunit tests

### DISCUSSION THREAD

**Discussion by eerison** [Codeception/Codeception#6748] [Q&A] [answered]
Is there anyway to run phpunit tests via codeception?

**Comment by burned42** [comment]
From my experience I'd say this should work just fine. I already had some occasions where I needed some specific assertions from symfony, similar to the test example at https://symfony.com/doc/current/mailer.html#write-a-functional-test which extends a test class from symfony that itself extends from phpunit and codeception executed the test just as every codeception test without any issues.

**Reply by eerison** [reply]
Hmmm but when I run bin/codeception,  it doesn't work for tests that use phpunit only.

Do  you do something else, or is it just work for you 🤔

**Reply by burned42** [reply]
I usually just run `vendor/bin/codecept run` which picks up those tests and executes them besides the codeception tests. 

For the filename I used `*Test.php` instead of `*Cest.php` which I usually use with codeception, the naming may be important.

I can't think of anything else special right now :thinking:

**Reply by eerison** [reply]
Well I just test and `codecept run`  doesn't execute phpunit tests :/

I get this error

> In Run.php line 495:                                                                                                       
> Suite 'MyTest.php' could not be found

**Reply by burned42** [reply]
@eerison I just tried it myself using the exact example "Code" and "Test Code" classes from https://phpunit.de/getting-started/phpunit-11.html and codeception is running that test without any issues:

```
$ vendor/bin/codecept run
Codeception PHP Testing Framework v5.1.2 https://stand-with-ukraine.pp.ua

Tests.Unit Tests (2) ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
✔ EmailTest: Can be created from valid email(0.01s)
✔ EmailTest: Cannot be created from invalid email(0.00s)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Time: 00:00.061, Memory: 10.00 MB

OK (2 tests, 2 assertions)
```

**Comment by eerison** [comment]
Hey Burned thank you to test it on your side <3

then the only thing different is

I am passing `test path` and I am using `phpunit 9`

maybe there is something wrong from my side, But I couldn't identify yet :/

**Reply by Naktibalda** [reply]
The error you got was about your command line parameters.
See examples at https://codeception.com/docs/reference/Commands#Run

That list is probably not excessive, but the way you want to specify a single test or file is incorrect.

**Reply by eerison** [reply]
Hey Nakti Thank you for reply

I just thought I could run like this  `codecept run tests/acceptance/MyCest.php`, it works for codeception tests, But not for phpunit ones :/

**Reply by Naktibalda** [reply]
I successfully executed https://github.com/Codeception/Codeception/blob/5.1/tests/unit/Codeception/StepTest.php
by running `./codecept run tests/unit/Codeception/StepTest.php`

Can you spot a difference between that file and your files?

**Reply by eerison** [reply]
just to make it simple I created another tests with

```php
<?php

declare(strict_types=1);

namespace crmi\tests\integration\middleware;

use PHPUnit\Framework\TestCase;

final class EmailTest extends TestCase
{
    public function testCanBeCreatedFromValidEmail(): void
    {
        $this->assertTrue(true);
    }
}
```

✅  /app $ php vendor/phpunit/phpunit/phpunit test/tests/integration/middleware/EmailTest.php 
[ERROR] /app $ php vendor/codeception/codeception/codecept run test/tests/integration/middleware/EmailTest.php 

codeception version:  5.1.2
phpunit version: 9.6.19 
php version: PHP 8.2.5

**Reply by Naktibalda** [reply, marked as answer]
Is your codeception.yml file located in app directory?
I expect tests path set to `test/tests`, is that right?
```yml
paths:
    tests: test/tests
```

Then there should be a test suite configuration file at `test/tests/integration.suite.yml`
Do you have such file?

**Reply by eerison** [reply]
Hey @Naktibalda 

it was the issue ... I have tested in other `tests` folder that contain this codeception config, and it worked.

thank you <3
